lib/game.rb

```diff
@@ -0,0 +1,105 @@
+class Game
+
+	attr_accessor :board, :player_1, :player_2
+
+
+	WIN_COMBINATIONS = [
+	[0, 1, 2], #top row
+	[3, 4, 5], #middle row
+	[6, 7, 8], #bottom row
+	[0, 3, 6], #left column
+	[1, 4, 7], #middle column
+	[2, 5, 8], #right column
+	[0, 4, 8], #top left-to-bottom-right
+	[2, 4, 6] #top right-to-bottom-left
+	]
+
+	def initialize( player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new )
+		@player_1 = player_1
+		@player_2 = player_2
+		@board = board
+	end
+
+	def current_player
+		board.turn_count.even? ? @player_1 : @player_2
+	end
+
+	def over?
+		draw? || won? 
+	end
+
+	def draw?
+		board.full? && !won?
+	end
+
+	def won?
+		WIN_COMBINATIONS.detect do |win_combo|
+      board.cells[win_combo[0]] == board.cells[win_combo[1]] &&
+      board.cells[win_combo[1]] == board.cells[win_combo[2]] &&
+      board.taken?(win_combo[0]+1) # add 1 to return number between 1-9
+    	end
+	end
+
+	def winner
+		if winning_combo = won?
+			the_winner = board.cells[winning_combo.first]
+		end
+	end
+
+	def turn
+		puts "To Make Your Move Pick a Number Between 1-9"
```

Comment:
I would refrain from putting inputs outputs here if we could redirect this to the cli somehow.